Feature/recruitment/models

Jiwoong0322 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #25 
- 작업 요약: recruitments의 Tag model 수정

## 📄 상세 내용
- [x] recruitments/models/tags.py의 name필드에 unique=True를 추가. 태그명이 중복되는 걸 방지.

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

closes #25

Jiwoong0322 committed:
1. tags.py의 name을 요구사항 정의서에서 명시된 대로 이름이 중복되는걸
   방지하기 위해 unique 적용
2. recruitment_attachments.py에 file_url을 URLFiled에서 FileFIeld로 바꿔
   다른 사용자가 첨부파일을 받아볼 수 있게 변경
3. 미디어 파일을 위한 설정을 base.py에 적용.

관련 이슈: #25

Jiwoong0322 committed:
💡 chore: recruitments_attachments.py 수정

1) commit후 checkout을 하려고 하니 recruitment_attachments에서
   error발생. 단순 줄바꿈 때문으로 보임.

관련 이슈: #25

Umdoong commented on apps/recruitments/models/recruitment_attachments.py:
```diff
@@ -13,8 +13,10 @@ class RecruitmentAttachment(BaseModel):
-    file_name = models.CharField(max_length=50, help_text="원본 파일명")
+    # URLField -> FileField 로 변경, upload_to 경로 지정
+    # FileField를 사용하여 사용자가 업로드한 '파일 자체'를 서버의 지정된 위치에 물리적으로 저장하는 역할. 파일이 저장될 '지정된 위치'는 setting에서 지정한다.
+    file_url = models.FileField(upload_to="recruitment_attachments/%Y/%m/%d/", help_text="첨부파일")
```
django-storage를 사용해서 S3 storage 설정을 통해 관리하겠다는 의미라면 이해하겠는데,
"recruitments/models/recruitments_attachments의 file_url을 사용자가 첨부파일을 업로드하는 것에 맞춰서 URL이 아닌 파일자체를 서버에서 관리하도록 URLField -> FileField로 변경"이라고 적어주셨기 때문에 정확히 알고 모델을 바꾸신 건지 의문이 듭니다.

Umdoong commented on apps/recruitments/models/tags.py:
```diff
@@ -4,7 +4,7 @@
 
 class Tag(BaseModel):
-    name = models.CharField(max_length=20, help_text="태그명")
+    name = models.CharField(max_length=20, unique=True, help_text="태그명")
```
tag 이름을 unique로 두는 건 좋은 것 같아요

Meoyoug commented on pyproject.toml:
```diff
@@ -8,7 +8,7 @@ authors = [
 requires-python = ">=3.12"
 dependencies = [
-    "django (>=5.2.1,<6.0.0)",
+    "django (>=5.2.5,<6.0.0)",
```
엇 장고버젼은 고정입니다 별도로 업데이트 안하시는게 좋아요

Jiwoong0322 replied:
poetry와 Django관련해서 shell에서 만지작 거리다가 저도 모르는새에 업데이트가 되버린거같습니다. 수정하였습니다.

Meoyoug commented on apps/recruitments/models/recruitment_attachments.py:
모델 변경 후에 마이그레이션을 진행하지 않으신 건지 마이그레이션 파일이 안올라와있네요?

Umdoong commented on config/settings/base.py:
MEDIA_URL은 dev.py, local.py에 이미 있습니다.

Meoyoug commented on apps/recruitments/models/recruitment_attachments.py:
```diff
@@ -13,8 +13,10 @@ class RecruitmentAttachment(BaseModel):
-    file_name = models.CharField(max_length=50, help_text="원본 파일명")
+    # URLField -> FileField 로 변경, upload_to 경로 지정
+    # FileField를 사용하여 사용자가 업로드한 '파일 자체'를 서버의 지정된 위치에 물리적으로 저장하는 역할. 파일이 저장될 '지정된 위치'는 setting에서 지정한다.
+    file_url = models.FileField(upload_to="recruitment_attachments/%Y/%m/%d/", help_text="첨부파일")
```
이필드의 경우 S3Uploader를 활용해서 직접 업로드 후 반환되는 S3 버킷 URL을 담을 필드라 
기조대로 유지해주시면 좋겠습니다

Jiwoong0322 replied:
네! URLField로 다시 적용하였습니다

Meoyoug commented on apps/recruitments/models/tags.py:
이거에 대해서만 마이그레이션 진행하고 PR다시올려주세요

Jiwoong0322 replied:
마이그레이션 진행하고 푸쉬했습니다.

Umdoong commented on pyproject.toml:
```diff
@@ -8,7 +8,7 @@ authors = [
 requires-python = ">=3.12"
 dependencies = [
-    "django (>=5.2.1,<6.0.0)",
+    "django (>=5.2.5,<6.0.0)",
```
큰 문제는 아니지만 종속성의 버전을 업그레이드 하신 이유를 달아주시면 더 좋을 것 같습니다.

Umdoong reviewed (changes requested):
코멘트 보시고 마이그레이션 후 다시 PR 올려주세용

Jiwoong0322 committed:
🚚 build: Django 버전 Develop과 동일하게 변경

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
poetry.lock, pyproject.toml의 Django 버전을 5.2.1로 변경했습니다.

Jiwoong0322 committed:
🚚 build: 마이그레이션 진행

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
base.py, recruitment_attachments.py를 되돌리고 tags.py마이그레이션 진행

3. 이슈가 있다면 아래에 연결하세요 ( 선택 )

관련 이슈: #25

Meoyoug reviewed (approved):
고생하셨습니다 머지할게요
